### PR TITLE
Add additional fields to Memcached Instance

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/terraform-providers/terraform-provider-google/issues/6907

For the field, memcachedNodes.state, do we keep it as string or Enum? what happens if I set it to Enum and google adds more values?

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
memcached: added memcacheVersion input and memcacheNodes output field to `google_memcache_instance` 
```

### CODE REVIEW

File: products/memcache/api.yaml [hunk 1]
```diff
@@ -118,6 +153,16 @@ objects:
...
+      - !ruby/object:Api::Type::Enum
+        name: memcacheVersion
+        description: |
+          The major version of Memcached software. If not provided, latest supported version will be used.
+          Currently the latest supported major version is MEMCACHE_1_5. The minor version will be automatically
+          determined by our system based on the latest supported minor version.
+        default_value: :MEMCACHE_1_5
+        values:
+          - :MEMCACHE_1_5
+          - :MEMCACHE_VERSION_UNSPECIFIED
```
Comment (danawillow) [inline products/memcache/api.yaml:165]: There isn't a good reason why a user would set MEMCACHE_VERSION_UNSPECIFIED, so I'd recommend omitting it.

File: products/memcache/api.yaml [hunk 2]
```diff
@@ -92,6 +93,35 @@ objects:
...
+            - !ruby/object:Api::Type::String
+              name: 'host'
+              description: Hostname or IP address of the Memcached node used by the clients to connect to the Memcached server on this node.
+            - !ruby/object:Api::Type::String
+              name: state
+              description: |
+                The major version of Memcached software. If not provided, latest supported version will be used.
+                Currently the latest supported major version is MEMCACHE_1_5. The minor version will be automatically
+                determined by our system based on the latest supported minor version.
+              # values:
```
Comment (danawillow) [inline products/memcache/api.yaml:122]: Your instinct was right here- it's not worth doing enums for output-only fields. There isn't really an enum type in Terraform- they're just strings with ValidateFuncs to ensure that bad values fail at plan-time. We've added it as a separate type to MM for convenience, but it only makes sense for fields that can actually be set.

File: products/memcache/api.yaml [hunk 3]
```diff
@@ -92,6 +93,35 @@ objects:
             - !ruby/object:Api::Type::String
               name: 'message'
               description: The message to be displayed to a user.
+      - !ruby/object:Api::Type::Array
+        name: 'memcacheNodes'
+        description: |
+          Additional information about the instance state, if available.
+        output: true
```
Comment (danawillow) [inline products/memcache/api.yaml:100]: You'll also need to add `output: true` for each of the nested fields (see https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3821-old..auto-pr-3821#diff-27973e511911c94f8104f416cb8733abR157-R192 for what it ends up looking like, with only the outer one being Computed and the rest being Optional).

File: products/memcache/api.yaml [hunk 4]
```diff
@@ -51,6 +51,7 @@ objects:
       guides:
         'Official Documentation':
           'https://cloud.google.com/memcache/docs/creating-instances'
+      api: 'https://cloud.google.com/memorystore/docs/memcached/reference/rest'
```
Comment (danawillow) [inline products/memcache/api.yaml:54]: Thank you!

File: products/memcache/api.yaml (hunk 2 above)
Comment (upodroid) [reply products/memcache/api.yaml:122]: fixed

File: products/memcache/api.yaml (hunk 1 above)
Comment (upodroid) [reply products/memcache/api.yaml]: fixed

**Pull request merged**